### PR TITLE
test: p2p_blockfilters tests for BIP157 config args

### PR DESCRIPTION
**Pull request by robot-dreams**
Based on #19720 

> Add tests for BIP157 configuration options in `p2p_blockfilters` as mentioned in [#19070 (review)](https://github.com/bitcoin/bitcoin/pull/19070#pullrequestreview-424449204). While here, optionally test with tighter `extra_args`, change an `info` log to `debug` as it doesn't describe a test, and fix up the message names in the header docstring.

### CODE REVIEW

File: test/functional/p2p_blockfilters.py [hunk 1]
```diff
@@ -47,11 +47,25 @@ def set_test_params(self):
...
+        err_msg = 'Error: Cannot set -peerblockfilters without -blockfilterindex.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=0', '-peerblockfilters'], err_msg)
+
+        self.log.info('Check that passing unknown -blockfilterindex type raises init error')
+        err_msg = 'Error: Unknown -blockfilterindex value foo.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=foo', '-peerblockfilters'], err_msg)
+
+        # Restart and restore default "chain" topology
+        self.start_node(0, extra_args=self.extra_args[0])
+        self.connect_nodes(1, 0)
```
Comment (jonatack) [inline test/functional/p2p_blockfilters.py:67]: ISTM this stopping, restarting, and connecting can be avoided per https://github.com/bitcoin/bitcoin/pull/19720/files#r499226085.

File: test/functional/p2p_blockfilters.py [hunk 2]
```diff
@@ -47,11 +47,25 @@ def set_test_params(self):
...
         ]
 
     def run_test(self):
+        self.stop_node(0)
+
+        self.log.info('Check that -peerblockfilters without -blockfilterindex raises init error')
+        err_msg = 'Error: Cannot set -peerblockfilters without -blockfilterindex.'
+        self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=0', '-peerblockfilters'], err_msg)
+
+        self.log.info('Check that passing unknown -blockfilterindex type raises init error')
```
Comment (jonatack) [inline test/functional/p2p_blockfilters.py:61]: nit: here, above line 57, and below line 212, I prefer the less-verbose "Test..." over "Check that..." and it's also probably the most frequently used term in the functional test logging. It's subjective though, apart from being less long.

File: test/functional/p2p_blockfilters.py [hunk 3]
```diff
@@ -78,7 +92,7 @@ def run_test(self):
         assert int(self.nodes[0].getnetworkinfo()['localservices'], 16) & NODE_COMPACT_FILTERS != 0
         assert int(self.nodes[1].getnetworkinfo()['localservices'], 16) & NODE_COMPACT_FILTERS == 0
 
-        self.log.info("get cfcheckpt on chain to be re-orged out.")
+        self.log.debug("get cfcheckpt on chain to be re-orged out.")
```
Comment (jonatack) [inline test/functional/p2p_blockfilters.py:95]: pico-nit, if you retouch maybe (a) s/get/Get/ and (b) can remove the EOL full stop per most frequent practice in the functional test logging

File: test/functional/p2p_blockfilters.py [hunk 4]
```diff
@@ -195,7 +209,7 @@ def run_test(self):
         computed_cfhash = uint256_from_str(hash256(cfilter.filter_data))
         assert_equal(computed_cfhash, stale_cfhashes[999])
 
-        self.log.info("Requests to node 1 without NODE_COMPACT_FILTERS results in disconnection.")
+        self.log.info("Check that requests without NODE_COMPACT_FILTERS results in disconnection.")
```
Comment (jonatack) [inline test/functional/p2p_blockfilters.py:212]: nit, here and line 106, can remove the full stop while changing this

**Review by jonatack — COMMENTED** [review]
Concept ACK, thanks for picking this up.

**Comment by jonatack** [comment]
Just found an odd edge case in manual testing when the node is already up:
```
$ ./src/bitcoind -blockfilterindex=1
Error: Unknown -blockfilterindex value 1.
```

**Comment by jonatack** [comment]
Just found an odd edge case in manual testing when the node is already up:
```
$ ./src/bitcoind -blockfilterindex=1
Error: Cannot obtain a lock on data directory /home/jon/.bitcoin. Bitcoin Core is probably already running.

$ ./src/bitcoind -blockfilterindex=basic
Error: Unknown -blockfilterindex value 1.
```

**Comment by jonatack** [comment]
Seems reproducible:
```python
self.log.info('Test passing two valid -blockfilter types raises init error')
err_msg = 'Error: Unknown -blockfilterindex value basic -blockfilterindex=1.'
self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=basic -blockfilterindex=1', '-peerblockfilters'], err_msg)
err_msg = 'Error: Unknown -blockfilterindex value basic -blockfilterindex=basic.'
self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=basic -blockfilterindex=basic', '-peerblockfilters'], err_msg)
err_msg = 'Error: Unknown -blockfilterindex value 1 -blockfilterindex=basic.'
self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=1 -blockfilterindex=basic', '-peerblockfilters'], err_msg)
err_msg = 'Error: Unknown -blockfilterindex value 1 -blockfilterindex=1.'
self.nodes[0].assert_start_raises_init_error(['-blockfilterindex=1 -blockfilterindex=1', '-peerblockfilters'], err_msg)
eerblockfilters'], err_msg)
```

**Comment by theStack** [comment]
May I ask what is the point of closing a PR with "Up for grabs" without even addressing its review comments? 
From my understanding the "Up for grabs" label is mainly used if a PR has been abandoned for a long time and the original author is not actively working on it anymore (that was not the case for #19720, the PR was not even 2 months old, had one ACK and unaddressed review comments).

**Comment by jonatack** [comment]
A combination of it not being on my priority list right now, not being merged despite favorable review (thank you), and my preference to not encumber the review stack with more than a few open PRs.

**Comment by theStack** [comment]
@robot-dreams: Are you still available/interested to work on this PR? Happy to review as soon as the necessary rebase happens.

**Comment by adamjonas** [comment]
Checked-in with @robot-dreams and he does not intend on moving forward with this. Closing.

**Comment by MarcoFalke** [comment]
Picked up in https://github.com/bitcoin/bitcoin/pull/24698

**Pull request closed without merging**